unyt/tests/test_array_functions.py

```diff
@@ -654,3 +654,15 @@ def test_trim_zeros():
     x1 = [0, 1, 2, 3, 0] * cm
     res = np.trim_zeros(x1)
     assert type(res) is unyt_array
+
+
+def test_any():
+    assert not np.any([0, 0, 0] * cm)
+    assert np.any([1, 0, 0] * cm)
+
+
+def test_append():
+    a = [0, 1, 2, 3] * cm
+    b = np.append(a, [4, 5, 6])
```

Comment:
shouldn’t this error out unless `a` is dimensionless?

Reply:
yes, but are we not treating (pure arrays / list / pick your poison) as dimensionless ?

Reply:
Yup, which is why this should error out, since `a` has non-dimensionless units.

Reply:
IMO, no. That would also be different from how ufuncs work.

Reply:
fair enough.
I reverted the previous behaviour but still fixed the validator function because it doesn't behave properly on the main branch for pure arrays